### ISSUE TITLE
(aws-ecs): Update Fargate service with new image tag using typescript cdk code

### ISSUE THREAD

**Issue by revanthalampally** [aws/aws-cdk#15300]
I'm trying to update my already created ECS service with a new task definition with typescript cdk But I do not find any ECS.FargateService.Update methods.
What is the best way to update my service in this case?

The reason I need to do this is:
I m trying to achieve Blue-Green deployment whenever my image tag changes in the service. Whenever I try to do a CDK deploy with a new image tag, I get:
`Resource handler returned message: "Invalid request provided: UpdateService error: Unable to update task definition on services with a CODE_DEPLOY deployment controller. Use AWS CodeDeploy to trigger a new deployment. (Service: AmazonECS; Status Code: 4`
After some reading, I found that CDK deploy tries to change the same task revision which won't work with the BLUE GREEN controller. 

Hence I thought of manually writing my own stack to update service consisting  of:
 1) Create a new task revision with a new image tag
 2) Update service manually to the new task revision
Basically, implementing whatever I do from the UI to trigger a Blue-Green Deployment.
But I am stuck in completing writing step 2 since there is not an option in the CDK code. Wondering if anyone has done this before?

### Environment

  - **CDK CLI Version:< 1.95.1 (build ed2bbe6)>
  - **aws-ecs: <1.97.0>

**Comment by revanthalampally** [comment]
@SoManyHs / @uttarasridhar : Can I know if there is any workaround? Or if I am doing anything wrong?

**Comment by revanthalampally** [comment]
Can anyone please reply to this?

**Comment by peterwoodworth** [comment]
Sorry no one has replied to this @revanthalampally,

ECS b/g deployments are really tricky for the CDK to handle, since under the hood CloudFormation's blue/green deployments have some limitations. Please read this thread for more info https://github.com/aws/aws-cdk/issues/1559#issuecomment-906580573, I'll be closing this issue in favor of that
